python/lsst/dax/imgserv/imageREST_v1.py

```diff
@@ -0,0 +1,226 @@
+    return _getimage_avail(request)
+
+@imageRESTv1.route("/capabilities", methods=["GET"])
+def getimage_capabilities():
+    return _getimage_capabilities(request)
+
+@imageRESTv1.route("/<db_id>", methods=["GET"])
+def getimage_sync(db_id):
+    return _getimage(request, db_id)
+
+@imageRESTv1.route("/<db_id>/async", methods=["POST"])
+def getimage_async(db_id):
+    return _getimage_async(request)
+
+def _getimage_avail(_req):
+    """Return availability status."""
+    fmt  = _req.accept_mimetypes.best_match(["application/json",
+        "text/html"])
+    if fmt == "text/html":
+        return "DAX Image Web Service <p> v0: service is accepting queries. <p>\
+                v1: service is accepting queries."
+    else:
+        json.dumps({
+            "availability": {
+                "api": [ "v0", "v1" ],
+                "v0": {
+                    "status": "dax_imgserv_v0 is accepting queries",
+                    'available': 'true'
+                },
+                "v1": {
+                    "status": "dax_imgserv_v1 is accepting queries",
+                    "available": "true"
+                }
+            }
+        })
+
+
+def _getimage_capabilities(_req):
+    """Return capabilities of this service."""
+    fmt  = _req.accept_mimetypes.best_match(["application/json",
+        "text/html"])
+    if fmt == "text/html":
+        return "Image Web Service <p> <a href='image/v0/DC_W13_Stripe82'> API v0 - DB: W13_sdss \
+                Interface: DAX <p> <a href='image/v1/DC_W13_Stripe82'>API v1 - \
+                DB: W13_sdss &nbsp; Interface: DAX </a>"
+    else:
+        json.dumps({
+            "capabilities": {
+                "api": [ "v0", "v1" ],
+                "v0": {
+                    "db": "W13_sdss",
+                    "url": "image/v0/DC_W13_Strip82",
+                    "interface": [ "DAX 0.0" ]
+                },
+                "v1": {
+                "db": "W13_sdss",
+                "url": "image/v1/DC_W13_Stripe82",
+                "interface": [ "SODA 1.1", "DAX 1.0" ]
+                }
+            }
+        })
+
+
+def _getimage(_req, db_id):
+    """Get the image per query request synchronously (default).
+    Parameters:
+        request the request object
+        db  image database string
+    """
+    ds = _req.args.get("ds")
+    if ds is None:
+        return None
+    w13db = _get_ds(ds.strip())
+    if w13db:
+        dispatcher = Dispatcher(current_app.config["DAX_IMG_CONFIG"])
+        params = _req.args.copy()
+        params["db"] = db_id
+        params["ds"] = ds
+        api = dispatcher.find_api(params)
+        try:
+            img_getter = image_open_v1(w13db, current_app.config)
+            image = api(img_getter, params)
+            if image is None:
+                return _image_not_found()
+            else:
+                return _file_response(image, "image_out.fits")
+        except:
+            # ToDo: handling error 
+            return None
```

Comment:
IMPORTANT: You should let the exception be raised or wrap it in a general exception. This must be changed. This will throw silent errors.

Reply:
Fixed by recognizing and throwing exceptions upwards for _error_handler() to capture.